### DISCUSSION TITLE
Training 7000 documents is taking more than 5 hours

### DISCUSSION THREAD

**Discussion by fabriziox1234** [explosion/spaCy#12851] [Help: Coding & Implementations] [answered]
As stated, I'm making an instance of the LDA model using multiple cores yet it still taking over 5 hours for the code to finish. As seen in the code, I'm trying to gather the coherence score for 5 different k. Any suggestions? I feel like this is awfully slow for what I'm trying to do

coherence = []

for k in range(5, 10):
    print('Round: '+str(k))
    lda_model = gensim.models.LdaMulticore(corpus=corpus,
                                            id2word=id2word,
                                            num_topics=k,
                                            workers=3,
                                            passes = 2,
                                            iterations = 2,
                                            chunksize = 1000,
                                            eval_every=None
                                            )
    
    cm = gensim.models.coherencemodel.CoherenceModel(model=lda_model, texts=corpus,
                                                   dictionary=id2word, coherence='c_v')
    coherence.append((k,cm.get_coherence()))

**Comment by shadeMe** [comment, marked as answer]
This appears to be a specific usage question about the `gensim` library. Please use their [support forum](https://groups.google.com/g/gensim) for such questions.
